lib/chef/knife/ssh.rb

```diff
@@ -181,18 +187,34 @@ def configure_session
         session_from_list(list)
       end
 
+      def get_prefix_attribute(node)
+        # Order of precedence for prefix
+        # 1) command line attribute
+        # 2) configuration file
+        # 3) nil
+        if node["prefix_cli"]
+          Chef::Log.debug("Using node attribute from CLI '#{config[:prefix]}' as the prefix: #{node["prefix_cli"]}")
+          node["prefix_cli"]
+        elsif Chef::Config[:knife][:prefix_attribute]
+          Chef::Log.debug("Using node attribute from config '#{Chef::Config[:knife][:prefix_attribute]}' as the prefix: #{node["prefix_config"]}")
+          node["prefix_config"]
+        else
+          nil
+        end
+      end
+
       def get_ssh_attribute(node)
         # Order of precedence for ssh target
         # 1) command line attribute
         # 2) configuration file
         # 3) cloud attribute
         # 4) fqdn
-        if node["config"]
-          Chef::Log.debug("Using node attribute '#{config[:attribute]}' as the ssh target: #{node["config"]}")
-          node["config"]
+        if node["ssh_cli"]
+          Chef::Log.debug("Using node attribute from CLI '#{config[:attribute]}' as the ssh target: #{node["ssh_cli"]}")
+          node["ssh_cli"]
         elsif Chef::Config[:knife][:ssh_attribute]
-          Chef::Log.debug("Using node attribute #{Chef::Config[:knife][:ssh_attribute]}: #{node["knife_config"]}")
-          node["knife_config"]
+          Chef::Log.debug("Using node attribute from config '#{Chef::Config[:knife][:ssh_attribute]}' as the ssh target: #{node["ssh_config"]}")
+          node["ssh_config"]
```

Comment:
this looks like it changed an external API on the node?

Reply:
`node` in this context is the response from the custom `query.search` call we make in `search_nodes`.